src/main.zig

```diff
@@ -2846,20 +2860,38 @@ fn fmtPathFile(
     if (stat.kind == .Directory)
         return error.IsDir;
 
-    const source_code = source_file.readToEndAllocOptions(
-        fmt.gpa,
-        max_src_size,
-        std.math.cast(usize, stat.size) catch return error.FileTooBig,
-        @alignOf(u8),
-        null,
-    ) catch |err| switch (err) {
-        error.ConnectionResetByPeer => unreachable,
-        error.ConnectionTimedOut => unreachable,
-        error.NotOpenForReading => unreachable,
-        else => |e| return e,
+    const source_code = blk: {
+        const source_code = source_file.readToEndAllocOptions(
+            fmt.gpa,
+            max_src_size,
+            std.math.cast(usize, stat.size) catch return error.FileTooBig,
+            @alignOf(u16),
+            null,
+        ) catch |err| switch (err) {
+            error.ConnectionResetByPeer => unreachable,
+            error.ConnectionTimedOut => unreachable,
+            error.NotOpenForReading => unreachable,
+            else => |e| return e,
+        };
+        source_file.close();
+        file_closed = true;
+        errdefer fmt.gpa.free(source_code);
+
+        // If the file starts with a UTF-16 BOM, translate it to UTF-8
+        if (mem.eql(u8, source_code[0..2], "\xff\xfe")) {
```

Comment:
The conversion code should be factored out into its own function, there's too much copy&paste. Something like:

```zig
fn convertTextToUtf8(allocator: *mem.allocator, input: *fs.File) []const u8
```

Reply:
Where should this function go ? Still somewhere in `main.zig` ?

Reply:
Yep

Reply:
This function was added, and I called it `readSourceFileToEndAlloc()`. It seemed more appropriate as it replaced `readToEndAllocOptions()`. Maybe the fact that it converts should be in the name ?

Can you review the implementation ?